instaloader/instaloader.py

```diff
@@ -312,15 +312,17 @@ def download_pic(self, filename: str, url: str, mtime: datetime,
                      filename_suffix: Optional[str] = None, _attempt: int = 1) -> bool:
         """Downloads and saves picture with given url under given directory with given timestamp.
         Returns true, if file was actually downloaded, i.e. updated."""
+        if filename_suffix is not None:
+            filename += '_' + filename_suffix
         urlmatch = re.search('\\.[a-z0-9]*\\?', url)
         file_extension = url[-3:] if urlmatch is None else urlmatch.group(0)[1:-1]
         if filename_suffix is not None:
             filename += '_' + filename_suffix
-        filename += '.' + file_extension
-        if os.path.isfile(filename):
-            self.context.log(filename + ' exists', end=' ', flush=True)
+        nominal_filename = filename + '.' + file_extension
+        if os.path.isfile(nominal_filename):
+            self.context.log(nominal_filename + ' exists', end=' ', flush=True)
             return False
-        self.context.get_and_write_raw(url, filename)
+        filename = self.context.get_and_write_raw(url, filename, get_extension=True)
```

Comment:
I think a good way to go for `download_pic()` would be:

1. Keep the logic that determines the filename with the extension retrieved from the URL and checks whether that file already exists
2. Determine an appropriate filename extension using the Content-Type
3. Check whether that file already exists, return False if so
4. Save the file, update its utime and return True

Reply:
> would requests to image server count into the quota that triggers "too many requests"?

No, they do not count into the quota.

Thanks! :)